src/main/java/cs2103/v15_1j/jimjim/ui/TaskEventRowFactory.java

```diff
@@ -40,33 +41,97 @@ public TaskEventRowFactory(DataLists masterList, DataLists displayList, GridPane
 		dateFmt = DateTimeFormatter.ofPattern("dd MMM yyyy");
 		dateTimeFmt = DateTimeFormatter.ofPattern("dd MMM h:mm a");
 	}
-
-	public void showTodayTaskEvents(){
+	
+	public void clear(){
 		rowNo = new Integer(-1);
 		pane.getChildren().clear();
-
-		Label headerLabel = new Label("Today");
-		headerLabel.getStyleClass().add("header");
-		pane.add(headerLabel, 0, ++rowNo, NO_OF_COLUMNS, 1);
-
-		int noOfTaskEvents = showTaskEventsOnDate(LocalDate.now());
-
-		Label tempEmptyLabel = new Label("");
-		pane.add(tempEmptyLabel, 0, ++rowNo, NO_OF_COLUMNS, 1);
-
-		if (noOfTaskEvents == 0){
-			pane.getChildren().remove(tempEmptyLabel);
-
-			Label noTaskEventsLabel = new Label("No Tasks or Events today");
-			noTaskEventsLabel.getStyleClass().add("no-overdue-label");
-			pane.add(noTaskEventsLabel, 0, ++rowNo, NO_OF_COLUMNS, 1);
+	}
+	
+	public void showAllTaskEvents(){
+		LocalDate currentDate = null;
+		int eventCounter = 0;
+		int deadlineTaskCounter = 0;
+		
+		List<Event> eventsList = masterList.getEventsList();
+		List<DeadlineTask> deadlineTasksList = masterList.getDeadlineTasksList();
+		
+		while(eventCounter < eventsList.size() 
+				&& deadlineTaskCounter < deadlineTasksList.size()){
+			Event nextEvent = eventsList.get(eventCounter);
+			LocalDate nextEventDate = nextEvent.getStartDateTime().toLocalDate();
+			DeadlineTask nextDeadlineTask = deadlineTasksList.get(deadlineTaskCounter);
+			LocalDate nextDeadlineTaskDate = nextDeadlineTask.getDateTime().toLocalDate();
+			
+			
+			if(!nextEventDate.isAfter(nextDeadlineTaskDate)){
+				if(currentDate == null){
```

Comment:
Just initialize `currentDate` to `LocalDate.MIN`, then you can eliminate this extra if else condition.